js/build-directory.js

```diff
@@ -1032,18 +1032,29 @@ DataDirectory.prototype.renderSearchResult = function( options, callback ){
   if (typeof callback === 'function') setTimeout(callback, 0);
 };
 
-DataDirectory.prototype.search = function( value ) {
-  var path = ':root > :has(';
+DataDirectory.prototype.search = function(search) {
+  var entries = [];
   var searchFields = this.config.search_fields;
+  this.data.forEach(function (entry) {
+    for (var i = 0; i < searchFields.length; i++) {
```

Comment:
@tonytlwu I want to return as soon I found a match on any field. With forEach you can't short-circuit it.